src/axom/inlet/docs/sphinx/advanced_types.rst

```diff
@@ -40,16 +54,52 @@ schema on that subtable, e.g.:
 
 The definition of a static ``defineSchema`` member function is not required, and is just used
 for convenience.  The schema definition for a class or struct could also be implemented as a
-free function for third-party types, or even in the same place as the sub-table declaration. 
+free function for third-party types, or even in the same place as the sub-table declaration.
 
-Arrays of user-defined types are also supported in Inlet.  First, use the ``addStructArray``
-function to create a subtable, then define the schema on that table:
+Accessing
+---------
+
+In order to have Inlet extract from the input file into a struct, it needs to know how that struct can be built.
+This is accomplished by a specializing of the struct ``FromInlet<T>`` for your type ``T``.  It must define a
+single member function with the signature ``T operator()(const inlet::Table&)``.  This function should return
+an instance of type ``T`` with its members populated with the corresponding fields in the input file.
+For the simple ``Car`` example whose schema is defined above, the specialization might look like:
 
 .. literalinclude:: ../../examples/user_defined_type.cpp
-   :start-after: _inlet_userdef_array_usage_start
-   :end-before: _inlet_userdef_array_usage_end
+   :start-after: _inlet_userdef_simple_frominlet_start
+   :end-before: _inlet_userdef_simple_frominlet_end
    :language: C++
 
+The ``Table::operator[]`` is used to extract fields from the input file and is automatically converted to
+the correct member variable types when the function's return value is constructed.  This conversion does
+not happen automatically for user-defined types.  If a ``Car`` object as defined above is located at the
+path "car" within the input file, it can be retrieved as follows:
+
+.. code-block:: C++
+
+  Car car = inlet["car"].get<Car>();
+
+**********************************
+Arrays and Dictionaries of Structs
+**********************************
+
+Arrays of user-defined types are also supported in Inlet.  
+
+Defining And Storing
+--------------------
+
+First, use the ``addStructArray`` function to create a subtable, then define the schema on that table:
```

Comment:
It would be helpful if the docs included a Lua / YAML example of input for this schema

Reply:
I've added a "fleet" as a `List[Car]` to both the `user_defined_type` example and to this docs page (replacing the references to the shapes examples entirely)